### ISSUE TITLE
[Feature Request]: Parse pipeline options only once at job submission.

### ISSUE THREAD

**Issue by tvalentyn** [apache/beam#26034]
### What would you like to happen?

Reparsing in https://github.com/apache/beam/blob/877833939360451da1f7bf7bd79532d3f88255db/sdks/python/apache_beam/runners/worker/sdk_worker_main.py#L102 can causes subtle bugs like: https://github.com/apache/beam/issues/25943

Instead, SDK can parse options at submission and workers can access options from portable pipeline representation.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner
